dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/net/RequestUniqueIdentifierTest.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.net
+
+import com.datadog.android.utils.forge.Configurator
+import fr.xgouchet.elmyr.annotation.RegexForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.annotation.StringForgeryType
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import okhttp3.MediaType
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class RequestUniqueIdentifierTest {
+
+    @RegexForgery("http(s?)://[a-z]+\\.com/\\w+")
+    private lateinit var fakeUrl: String
+
+    @RegexForgery("x-[a-z]+/[a-z]+")
+    private lateinit var fakeContentType: String
+
+    @StringForgery(StringForgeryType.ALPHABETICAL)
+    private lateinit var fakeBody: String
```

Comment:
is this always going to be a non - empty string ?

Reply:
Yes